plugins/database/src/mid_level_routines.cpp

```diff
@@ -1600,17 +1600,17 @@ int checkObjIdByTicket(const char* dataId,
 
     iUsesLimit = atoi( usesLimit );
     if ( iUsesLimit > 0 ) {
-        iUsesCount = atoi( usesCount );
-        if ( iUsesCount > iUsesLimit ) {
-            return CAT_TICKET_USES_EXCEEDED;
-        }
 
         intDataId = atoll( dataId );
 
         // Don't update a second time if this id matches the last one.
         // Given this function can be called multiple times within the same operation, checking to
         // see if the data id is different keeps the server from updating the ticket information multiple times.
         if ( previousDataId2 != intDataId ) {
+            iUsesCount = static_cast<int>(strtol(usesCount, nullptr, 0));
```

Comment:
Need to set errno to 0 before calling strtol().
Need to check errno too.

See https://en.cppreference.com/w/c/string/byte/strtol

Reply:
What should I do in the case `errno` is nonzero after the call?
1. Log a message (nothing else in `mid_level_routines.cpp` logs without the switch being enabled, so this is maybe weird?)
2. Return `CAT_TICKET_USES_EXCEEDED`
3. All of the above

Reply:
...or 4, something else?

Reply:
Log a message and return an error.

We shouldn't return `CAT_TICKET_USES_EXCEEDED` unless that's what happened. Perhaps one of the following could work:
- SYS_INTERNAL_ERR
- SYS_LIBRARY_ERROR
- Does rodsErrorTable.h define any conversion/parsing error?

If we don't have an error code that describes the situation well, we can talk about introducing a new one.